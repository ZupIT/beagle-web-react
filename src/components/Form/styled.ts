import styled from 'styled-components'

export const StyledForm = styled.form`
  position: relative;
  width: 500px;
`

export const LoadingPanel = styled.div<{ isVisible: boolean }>`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.6);
  opacity: ${({ isVisible }) => isVisible ? 1 : 0};
  pointer-events: ${({ isVisible }) => isVisible ? 'inherit' : 'none'};
  transition: opacity 0.3s;
  display: flex;
  align-items: center;
  justify-content: center;
`
