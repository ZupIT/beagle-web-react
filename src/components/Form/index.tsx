/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC } from 'react'
import Spinner from '../Spinner'
import { BeagleDefaultComponent } from '../types'
import withTheme from '../utils/withTheme'
import { StyledForm, LoadingPanel } from './styled'

export interface FormInterface extends BeagleDefaultComponent {
  isLoading?: boolean,
  onSubmit?: () => void,
  onReset?: () => void,
}

const Form: FC<FormInterface> = ({
  isLoading = false,
  onSubmit,
  onReset,
  style,
  className,
  children,
}) => {
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    if (onSubmit) onSubmit()
  }

  return (
    <StyledForm onSubmit={handleSubmit} onReset={onReset} style={style} className={className}>
      {children}
      <LoadingPanel isVisible={isLoading}>
        <Spinner />
      </LoadingPanel>
    </StyledForm>
  )
}

export default withTheme(Form)
