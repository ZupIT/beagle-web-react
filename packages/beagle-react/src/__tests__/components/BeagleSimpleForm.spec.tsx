/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import * as React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow, mount } from 'enzyme'
import Form from '../../components/BeagleSimpleForm'

let wrapper: any
let onSubmitMock: any

configure({ adapter: new Adapter() })
beforeAll(() => {
  onSubmitMock = jest.fn(()=>'submited')
  wrapper = mount(<Form onSubmit={onSubmitMock}/>)
})

test('Beagle snapshot SimpleForm', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should call on submit handler', () => {
  wrapper.simulate('submit')
  expect(onSubmitMock).toHaveBeenCalledTimes(1)
})

