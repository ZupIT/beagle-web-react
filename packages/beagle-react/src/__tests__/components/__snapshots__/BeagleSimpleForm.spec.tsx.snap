// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot SimpleForm 1`] = `
<ComponentWithTheme
  onSubmit={[MockFunction]}
>
  <Form
    className=""
    onSubmit={[MockFunction]}
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <ComponentWithTheme
        disabled={false}
        placeholder="Testing"
        readOnly={false}
        type="TEXT"
        value="Teste"
      >
        <BeagleInput
          className=""
          disabled={false}
          placeholder="Testing"
          readOnly={false}
          type="TEXT"
          value="Teste"
        >
          <styled.input
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Testing"
            readOnly={false}
            style={Object {}}
            type="TEXT"
            value="Teste"
          >
            <input
              className="sc-gsTCUz cbPgpe"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Testing"
              readOnly={false}
              style={Object {}}
              type="TEXT"
              value="Teste"
            />
          </styled.input>
        </BeagleInput>
      </ComponentWithTheme>
    </form>
  </Form>
</ComponentWithTheme>
`;
